### DISCUSSION TITLE
Per-element weighed loss function for reactive events

### DISCUSSION THREAD

**Discussion by mbocus** [mir-group/nequip#153] [Q&A] [answered]
Dear NequIP users/developers,

I was interested in training a potential to describe a chemical reaction, in a similar fashion as the formate decomposition on Cu reported in the original manuscript. As in that case, I have a large unbalance between reacting atoms and spectating atoms, therefore I would also like to use a per-element weighed loss function, where the reactive atoms receive a larger weight. However, I wasn't able to find an example concerning the syntax required in the input file. Ideally, having N atoms of type A (say copper atoms), I would like to be able to select M<N of them (the ones actually involved in the reaction) and increase their weight in the loss function compared to the others. Is that possible with the current NequIP implementation? 

Thanks in advance for your help!
Massimo

**Comment by nw13slx** [comment, marked as answer]
Hi Massimo,

Please use the statement below.

```
loss_coeffs:
  forces:
  - 1.0
  - PerSpeciesMSELoss
  total_energy:
  - 1.0
  - PerAtomMSELoss
```

I don't remember whether I have implemented the method to allow user-specified weights. Let me know whether this flag works first. We also have other unpublished implementations to improve the training of our formate Cu system. Please email me (lixinsun@g.harvard.edu) if you are interested in using them early.

Best,
Lixin

**Reply by mbocus** [reply]
Dear Lixin, 

Thank you very much for your kind reply, I am going to try your suggestion and, based on the results, see if I could further benefit from other type of weights.

Best regards,
Massimo
